File: src/temporal_learn/process2/starter.py
import asyncio
import uuid

from temporalio.client import Client

from .models import ProductInput
from .workflow import ProductOnboardingWorkflow

TASK_QUEUE = "product-onboarding-tq"


async def main():
    client = await Client.connect("localhost:7233")
    product_input = ProductInput(
        request_id=str(uuid.uuid7()),
        sku="SKU-12345",
        name="My Product",
        category="Shoes",
        currency="TWD",
        price=1990,
        visible=True,
    )
    handle = await client.start_workflow(
        ProductOnboardingWorkflow.run,
        product_input,
        id=f"product-{uuid.uuid7()}",
        task_queue=TASK_QUEUE,
    )
    print(f"WorkflowId: {handle.id}")


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/src/process2/starter.ts
import { Client, Connection } from '@temporalio/client';
import { randomUUID } from 'crypto';
import { TASK_QUEUE, TEMPORAL_URL, type ProductInput } from './models';
import { productOnboardingWorkflow } from './workflow';

async function main() {
  const connection = await Connection.connect({ address: TEMPORAL_URL });
  const client = new Client({ connection });
  const productInput: ProductInput = {
    requestId: randomUUID(),
    sku: 'SKU-12345',
    name: 'My Product',
    category: 'Shoes',
    currency: 'TWD',
    price: 1990,
    visible: true,
  };

  const handle = await client.workflow.start(productOnboardingWorkflow, {
    args: [productInput],
    taskQueue: TASK_QUEUE,
    workflowId: `product-ts-${randomUUID()}`,
  });

  console.log(`WorkflowId: ${handle.workflowId}`);
  await connection.close();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
